testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/input/MiniAppInputActivity.kt

```diff
@@ -89,11 +92,32 @@ class MiniAppInputActivity : MenuBaseActivity(), PreloadMiniAppWindow.PreloadMin
     }
 
     private fun displayMiniApp() = when(display) {
-        is InputDisplay.AppId -> preloadMiniAppWindow.initiate(null, display.input.trim(), this)
+        is InputDisplay.AppId -> {
+            initiatePreloadScreen(display.input.trim())
+        }
         is InputDisplay.Url -> MiniAppDisplayActivity.startUrl(this, display.input.trim())
         is InputDisplay.None -> {}
     }
 
+    private fun initiatePreloadScreen(miniAppId: String) {
+        val viewModel: MiniAppInputViewModel = ViewModelProvider.NewInstanceFactory().create(MiniAppInputViewModel::class.java)
+                .apply {
+                    // observe version id
+                    miniAppVersionId.observe(this@MiniAppInputActivity, Observer {
+                        preloadMiniAppWindow.initiate(
```

Comment:
I can pass a valid versionId from input after observing it in order to fix MINI-2941.